### PR TITLE
Improvements

### PR DESCRIPTION
**Pull request by andreypopp**
- Do not reverse attributes at runtime, fix order in ppx. This also makes it so nodes created without ppx have order of attributes preserved, instead of reversed.
- General code cleanup
- Reuse element repr for Debug

### CODE REVIEW

File: dune-project [hunk 1]
```diff
@@ -18,13 +17,12 @@
  (description "html_of_jsx is a JSX transformation that allows you to write HTML declaratively.")
  (depends
   (ocaml (>= 4.14))
-  (reason (>= 3.10.0))
   (ppxlib (> 0.23.0))
-  melange
```
Comment (davesnx) [inline dune-project:23]: melange should be added, since we have a melange version of it

File: docs/index.mld [hunk 2]
```diff
@@ -4,14 +4,14 @@
...
-{b html_of_jsx} is a JSX transformation to write HTML declaratively in {{:https://ocaml.org OCaml}, {{:https://reasonml.github.io} Reason} and {{:https://github.com/ocaml-mlx/mlx}mlx}.
+{b html_of_jsx} is a JSX transformation to write HTML declaratively in {{:https://ocaml.org}OCaml}, {{:https://reasonml.github.io}Reason} and {{:https://github.com/ocaml-mlx/mlx}mlx}.
 
 This library was extracted from {{: https://github.com/ml-in-barcelona/server-reason-react} server-reason-react} and later simplified to work only with HTML5.
 
 {1 Installation}
 
 {[
-opan install html_of_jsx
+opam install html_of_jsx
```
Comment (davesnx) [inline docs/index.mld:14]: lol

File: lib/JSX.ml [hunk 3]
```diff
@@ -62,60 +57,47 @@ let list arr = List arr
...
-      | Null -> Null
-      | String str -> String str
-      | Unsafe str -> Unsafe str
-      | Node { tag; attributes; children } ->
-          Node
-            { tag; attributes; children = List.map to_debug_element children }
-      | List list -> List (List.map to_debug_element list)
-    in
-    to_debug_element el
+  let view element = element
```
Comment (davesnx) [inline lib/JSX.ml:102]: 👍🏼

File: lib/dune [hunk 4]
```diff
@@ -1,5 +1,3 @@
 (library
- (name html_of_jsx)
- (wrapped false)
- (modes :standard melange)
```
Comment (davesnx) [inline lib/dune:4]: We might want to keep melange here as well

**Review by davesnx — APPROVED** [review]
Nice improvements

**Comment by andreypopp** [comment]
re: removing melange, see the commits, I've pushed one marked as PROPOSAL to
remove melange as a dependency.

The rational is that it's not super often you need to generate raw strings in
browser and maybe it's better to create `dom_of_jsx` ppx that would generate
DOM directly?

**Comment by davesnx** [comment]
Yes, we can have a html_of_jsx.melange package in here.

**Comment by andreypopp** [comment]
> Yes, we can have a html_of_jsx.melange package in here.

How should I proceed with this PR? If there are not known users — I suggest to remove melange support for now and then reinstate it as a separate package generating DOM elements.

re: CI failures — not sure why this is failing, doesn't seem related to this PR, now?

**Comment by davesnx** [comment]
To fix the build, could you add the restriction for tiny_httpd to be 0.16?

Happy to merge it after.

**Pull request merged**